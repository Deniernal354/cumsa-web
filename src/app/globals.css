@import "tailwindcss";

:root {
  --background: #f8fafc;
  --foreground: #0f172a;
  --surface: #ffffff;
  --muted: #64748b;
  --navy: #2a7487;
  --blue: #5aced8;
  --red: #c73221;
  --header-top: #123a66;
}

body {
  background: var(--background);
  color: var(--foreground);
  font-family: Arial, Helvetica, sans-serif;
}

a {
  text-underline-offset: 4px;
}

.link-cta {
  color: var(--blue);
  font-weight: 700;
  text-decoration: underline;
  text-underline-offset: 3px;
}

.link-cta:hover {
  color: var(--red);
}

.btn-cozy {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  padding: 0.5rem 1rem;
  border-radius: 0.6rem;
  font-weight: 600;
  font-size: 0.9rem;
  transition:
    transform 150ms ease,
    box-shadow 150ms ease,
    opacity 150ms ease;
}

.btn-cozy:hover {
  transform: translateY(-1px);
  box-shadow: 0 6px 16px rgba(0, 0, 0, 0.08);
}

.btn-primary {
  background: var(--blue);
  color: white;
}

.btn-outline {
  background: var(--surface);
  color: var(--foreground);
  border: 1px solid rgba(0, 0, 0, 0.12);
}

:focus-visible {
  outline: 2px solid color-mix(in oklab, var(--blue) 70%, white);
  outline-offset: 2px;
}

.container-base {
  margin-left: auto;
  margin-right: auto;
  max-width: 80rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

@media (min-width: 640px) {
  .container-base {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media (min-width: 1024px) {
  .container-base {
    padding-left: 2rem;
    padding-right: 2rem;
  }
}

.transition-base {
  transition: all 200ms ease-out;
}

.card {
  background: var(--surface);
  border: 1px solid rgba(0, 0, 0, 0.12);
  border-radius: 0.75rem;
}

.title {
  color: var(--red);
}

.tint-blue {
  filter: invert(28%) sepia(88%) saturate(3612%) hue-rotate(211deg) brightness(93%) contrast(98%);
}

.logo-mask-blue {
  background-color: var(--blue);
  -webkit-mask-image: url("/CU-MSA-LOGO-blank-2.png");
  mask-image: url("/CU-MSA-LOGO-blank-2.png");
  -webkit-mask-repeat: no-repeat;
  mask-repeat: no-repeat;
  -webkit-mask-position: center;
  mask-position: center;
  -webkit-mask-size: contain;
  mask-size: contain;
  display: inline-block;
}
