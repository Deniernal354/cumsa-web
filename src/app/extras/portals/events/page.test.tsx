import { test, expect } from "@jest/globals";
import { render } from "@testing-library/react";
import EventsPortalEntry from "./page";

test("events portal entry snapshot", () => {
  const { asFragment } = render(<EventsPortalEntry />);
  expect(asFragment()).toMatchSnapshot();
});
