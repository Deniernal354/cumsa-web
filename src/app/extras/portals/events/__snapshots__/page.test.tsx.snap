// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`events portal entry snapshot 1`] = `
<DocumentFragment>
  <div
    class="text-center space-y-8"
  >
    <div>
      <div
        class="text-8xl mb-6"
      >
        ðŸšª
      </div>
      <h1
        class="text-4xl font-extrabold text-[var(--blue)] mb-4"
      >
        Events Portal
      </h1>
      <p
        class="text-xl text-gray-600 max-w-2xl mx-auto leading-relaxed"
      >
        Welcome to the Events Management Portal. This is a secure area for authorized users to create, edit, and manage community events.
      </p>
    </div>
    <div
      class="bg-amber-50 border border-amber-200 rounded-xl p-6 max-w-md mx-auto"
    >
      <div
        class="flex items-center gap-3 mb-3"
      >
        <span
          class="text-2xl"
        >
          ðŸ”’
        </span>
        <h3
          class="font-semibold text-amber-800"
        >
          Authentication Required
        </h3>
      </div>
      <p
        class="text-sm text-amber-700 mb-4"
      >
        You need valid credentials to access the portal management features.
      </p>
      <a
        class="btn-cozy btn-primary w-full inline-block text-center"
        href="/extras/portals/events/login"
      >
        Login to Portal
      </a>
    </div>
    <div
      class="pt-6"
    >
      <p
        class="text-gray-500 text-sm mb-3"
      >
        Not authorized?
      </p>
      <div
        class="flex gap-4 justify-center"
      >
        <a
          class="btn-cozy btn-outline"
          href="/events"
        >
          View Public Events
        </a>
        <a
          class="text-[var(--blue)] hover:text-[var(--red)] underline text-sm py-3"
          href="/contact-us"
        >
          Request Access
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
